ocaml/xapi/xapi_observer.ml

```diff
@@ -124,6 +124,8 @@ module Component = struct
         "xenopsd"
     | Xapi_clusterd ->
         "xapi-clusterd"
+    | SMApi ->
+        "smapi"
```

Comment:
I think we should really have something like
```
module Component
  module Name
    xapi = "xapi"
    xenopsd = "xenopsd"
    xapi_clusterd = "xapi-clusterd"
    smapi = "smapi"
  end
...
```
and then reuse these constants Component.Name.xapi, Component.Name.xenopsd etc instead of strings throughout the code.

The same thing with Tracing.Endpoint.Name.bugtool = "bugtool", as I think @Vincent-lau  suggested in another comment.

Reply:
depends on how heavily we need to use the òf_string` function. If it's only once or twice while interfacing with the external world, then this should be fine. Otherwise we might need to consider something else